codegen/src/main/kotlin/software/amazon/smithy/rust/codegen/smithy/protocols/parse/JsonParserGenerator.kt

```diff
@@ -196,10 +196,13 @@ class JsonParserGenerator(
         )
     }
 
-    override fun serverInputParser(operationShape: OperationShape): RuntimeType {
-        val inputShape = operationShape.inputShape(model)
+    override fun serverInputParser(operationShape: OperationShape): RuntimeType? {
         val includedMembers = httpBindingResolver.requestMembers(operationShape, HttpLocation.DOCUMENT)
-        val fnName = symbolProvider.deserializeFunctionName(inputShape)
+        if (includedMembers.isEmpty()) {
+            return null
+        }
+        val inputShape = operationShape.inputShape(model)
+        val fnName = symbolProvider.deserializeFunctionName(operationShape)
```

Comment:
@rcoh remember the `unused mut` problem we discussed. It turned out to be a bug where we where generating deserializers for input without a JSON body. The protocol tests are AMAZING and allowed me to catch that!